src/deepsparse/transformers/pipelines/text_classification.py

```diff
@@ -267,3 +290,27 @@ def route_input_to_bucket(
         )
         input_seq_len = max(map(len, tokens["input_ids"]))
         return TransformersPipeline.select_bucket_by_seq_len(input_seq_len, pipelines)
+
+
+def _get_top_k(top_k: int, return_all_scores: bool, num_labels: int) -> int:
+    # handle deprecations of return_all_scores
+    if top_k <= 0:
+        raise ValueError(f"text_classification top_k must be > 0. found {top_k}")
+    if return_all_scores and top_k not in [1, num_labels]:
+        # top_k is not set to default, but does not match num_labels
+        # for return_all_scores
+        raise ValueError(
+            f"Mismatch: return_all_scores set to {return_all_scores} with "
+            f"{num_labels} in model config, but top_k set to {top_k}. "
+            "return_all_scores is deprecated, set top_k instead"
+        )
+
+    if top_k == 1 and return_all_scores:
```

Comment:
It feels a bit counter-intuitive to say an argument is deprecated (v.s. going to be deprecated) while it affects the flow of logic. Do we have a standardized way for handling deprecation?

Reply:
generally we continue support the best we can while raising a warning that will remind users (internal and external) to update their usage